### DISCUSSION TITLE
Query regarding Bulk insert

### DISCUSSION THREAD

**Discussion by VijayDeyTR** [milvus-io/milvus#21835] [Q&A and General discussion]
Hi,

I am trying to use the bulk insert api via files - https://milvus.io/docs/bulk_insert.md

Do these files need to be first published on the object storage, in our case we use s3?

**Comment by xiaofan-luan** [comment]
> Hi,
> 
> I am trying to use the bulk insert api via files - https://milvus.io/docs/bulk_insert.md
> 
> Do these files need to be first published on the object storage, in our case we use s3?

exactly ~
current bulk insert support files less than 1G, but we are working on expanding it to 10G+

**Reply by VijayDeyTR** [reply]
So the process to use bulk insert is something like, could you please confirm this?
1. Create and copy files to the object storage
2. Invoke bulk insert, specifying file names copied as part of step 1
3. Monitor the asynchronous tasks for the insertion results

**Reply by xiaofan-luan** [reply]
> So the process to use bulk insert is something like, could you please confirm this?
> 
> 1. Create and copy files to the object storage
> 2. Invoke bulk insert, specifying file names copied as part of step 1
> 3. Monitor the asynchronous tasks for the insertion results

Exactly, we will change the bulkload process to make it more clear in next release:
1. prepare files and put it into same S3/Minio
2. call bulkload API
3. check bulkload done
4. call index interface, this will ensure bulkloaded data is indexed(Optional)
5. call load refresh, this will ensure bulkloaded data can be searched (Optional)
